integration/sql/impl/src/context/mod.rs

```diff
@@ -371,15 +369,31 @@ impl<'a> Context<'a> {
         stack.push(ancestor.clone());
 
         while let Some(current) = stack.pop() {
-            let column_ancestors = old.column_ancestry.get(&current);
+            let column_ancestors = old
+                .column_ancestry
+                .get(&current)
+                // try to find by name in case of column without origin
+                // this can happen when selecting from subqery without alias
+                .or_else(|| {
+                    old.column_ancestry.iter().find_map(|(col, anc)| {
+                        if col.name == current.name && current.origin.is_none() {
+                            Some(anc)
+                        } else {
+                            None
+                        }
+                    })
+                });
             if column_ancestors.is_none() {
                 result.push(current.clone());
                 continue;
             }
 
             if let Some(ancestors) = column_ancestors {
-                for ancestor in ancestors {
-                    stack.push(ancestor.clone());
+                for anc in ancestors {
+                    // avoid infinite loop
+                    if anc != &current {
+                        stack.push(anc.clone());
+                    }
```

Comment:
Add some counter, to fix multi-element loop if it happens - we've seen it can happen a few times.

Reply:
Added.